src/main/java/jenkins/metrics/api/Metrics.java

```diff
@@ -451,6 +441,11 @@ public void run() {
                             }
                         } catch (InterruptedException e) {
                             e.printStackTrace(l.fatalError("aborted"));
+                        } catch (Exception e) {
+                            logger.log(Level.SEVERE, "Error running " + HealthChecker.class.getName(), e);
+                            if (l != null) {
+                                e.printStackTrace(l.fatalError(e.getMessage()));
+                            }
```

Comment:
It seems that the general policy in this plugin is to send messages to the `logger` if `l` (`StreamTaskListener`) is not available, is there some reason to use both here?